app/src/main/java/org/connectbot/transport/SSH.java

```diff
@@ -439,6 +440,7 @@ private void finishConnection() {
 
 	@Override
 	public void connect() {
+		connecting = true;
```

Comment:
It might be easier to follow if you extract this out to another function like:

```
public void connect() {
    connecting = true;
    try {
        connectInternal();
    } finally {
        connecting = false;
    }
}

private void connectInternal() {
    // previous contents of connect() ...
}
```